packages/core/admin/admin/src/components/Providers/index.js

```diff
@@ -64,15 +65,17 @@ const Providers = ({
                   settings={settings}
                 >
                   <LibraryProvider components={components} fields={fields}>
-                    <LanguageProvider messages={messages} localeNames={localeNames}>
-                      <AutoReloadOverlayBlockerProvider>
-                        <OverlayBlocker>
-                          <GuidedTour>
-                            <Notifications>{children}</Notifications>
-                          </GuidedTour>
-                        </OverlayBlocker>
-                      </AutoReloadOverlayBlockerProvider>
-                    </LanguageProvider>
+                    <CustomFieldsProvider customFields={customFields}>
```

Comment:
I'm wondering if this is really needed ?

We have the StrapiAppProvider that centralizes everything so you could just build a useCustomFields hook that will get them from the Strapi App instead of adding more Providers 🤔

Reply:
I think we did it this way because we wanted to mimic how components and fields were exposed in a separate LibraryProvider. But adding it to StrapiAppProvider sounds good 👍

Reply:
I see :) It goes with my old comment on passing this instead of each element we could pass the app to the app provider so everything would be available here it would be very powerful when extending the API. We can keep it this way and change the hook when we decide